### DISCUSSION TITLE
Redis 사용 여부에 대해

### DISCUSSION THREAD

**Discussion by shkisme** [BDD-CLUB/01-doo-re-back#34] [논의 사항]
초대 링크를 구현하며 Redis가 필요하다고 생각합니다.

1. 유효기간 1일의 TTL이 있다.
2. 따라서 DB에 저장할만 한 리소스가 아니다. 1일 후 휘발되어야 한다. (1일이 지나면 더이상 의미없는 자원이 됨)

처음에는 상태를 가지지 않게, 링크를 날짜 + teamId로 암호화하고, 그것을 복호화 하여 비교할까 생각했었습니다. 그런데 24시간 이내에 생성된 링크가 있는지 확인하고, 없다면 링크를 생성해야 할 것 같아서 DB에 링크 정보를 저장하기보다는 Redis를 활용하는 편이 좋겠다고 결론냈습니다.

**Comment by JJimini** [comment]
저는 Redis를 주로 사용해왔고 이번에도  Redis 사용에 동의하는데요, 이유는 다음과 같습니다!
- 엑세스가 빈번한 데이터를 다른 DB보다 빠르고 가벼운 인메모리 DB인 Redis를 통해 처리
- ex) 빈번한 로그인 만료로 인해 사용자의 이용 흐름을 방해하지 않아 편의성이 좋아짐

**Reply by 02ggang9** [reply]
저도 redis 사용에 동의 합니다. 

@shkisme, @JJimini 님 말씀처럼 짧은 유효기간을 가지고 있고 빈번하게 접근되는 데이터는 디스크에 저장하기 보다는 메모리에 저장하는게 효율적이라고 생각합니다. 

만약 redis를 사용하지 않으면 초대 링크 데이터가 만료 되었는지 확인하고 삭제하는 코드(유사 배치)가 추가로 작성되어야 하고 수시로 동작시켜야 하기 때문에 서버에 부담이 갈 것 같습니다. 또, redis 도입은 간단하고 가볍다고 생각해서 redis 도입에 찬성입니다~

**Comment by amaran-th** [comment]
혹시 어떤 포인트에서 Redis 사용을 고려하게 된 건지 알 수 있을까요?
새로운 기술을 도입하는 것은 그만큼 인프라가 복잡해지고 관리 난이도가 높아질 수 있는 위험성이 있기 때문에, '두레 서비스'에 Redis가 필요한 이유가 명확해야 한다고 생각합니다.

**Comment by lcqff** [comment]
언젠가는 사용
- 로그인은 일단 논외로
- 초대 코드와 같은 부분에서는 사용  => 이후 해당 부분 개발시 추가논의
